Feature: Landing page for Shop

0duaht opened the pull request:
Why?:
Homepage for guests/users visiting the shop site.

This addresses the need by:
Designing views and adding a controller to render the landing page of the shop website.

Header and Footer are seperated into partials for easy use with other pages.

Adding products and categories to be shown on homepage as an array from the controller.

Pages were designed with the Materialize CSS Framework.

[Finishes #108000292]

Commit:
Feature: Landing page for Shop

Why?:
Homepage for guests/users visiting the shop site.

This addresses the need by:
Designing views and adding a controller to render the landing page of the shop website.

Header and Footer are seperated into partials for easy use with other pages.

Adding products and categories to be shown on homepage as an array from the controller.

Pages were designed with the Materialize CSS Framework.

[Finishes #108000292]

ChigboIO commented on app/assets/stylesheets/landing.scss:
```diff
@@ -0,0 +1,173 @@
+.medium {
+  height: 45em;
+  width: 100%;
+  background: url("http://i.imgur.com/CShEgXJ.jpg") no-repeat;
```
Is there another way we can refactor this considering what `houndci` is saying?

ChigboIO commented:
We could remove the trailing white spaces that `houndci` is reporting

codesword commented:
@andela-toduah Did you check for style violations using rubocop before pushing?

ChigboIO commented:
Your `.erb` views are not following the two spaces indentation.

ChigboIO commented on app/assets/stylesheets/landing.scss:
```diff
@@ -0,0 +1,173 @@
+}
+
+.container{
+  width: 100%;
```
I think you should specifically set the container for the background image to 100%, as this will affect other containers elsewhere

ChigboIO commented on app/controllers/landing_controller.rb:
```diff
@@ -0,0 +1,55 @@
+    @products = [
+      [
+        { image: "http://i.imgur.com/QsgYAlQ.png",
+          text: "Sling-Back Stilletos", price: "N5000" 
```
Each `key` - `value` pair should be on their separate lines `(price: "N5000")`

ChigboIO commented on app/views/landing/_header.html.erb:
```diff
@@ -0,0 +1,54 @@
+		<div class="top-bar valign-wrapper">
+			<div class="container-trim">
+				<div class="left">
+					<span class="white-text small-text">FAST & FREE SHIPPING ON ALL ORDERS - CALL US : 08020000000</span>
```
I think it would be better to use `&amp;` here, instead of `&`

ChigboIO commented on app/views/landing/_header.html.erb:
```diff
@@ -0,0 +1,54 @@
+			<li>Electronics</li>
+			<li>Home and Kitchen</li>
+			<li>Baby, Kids and Toys</li>
+		</ul>
```
You already have these categories in the `@categories` variable defined in the controller. It would be better to use it here since these are going to be coming dynamically anyways.

0duaht replied:
Not every page will run my landing controller, as that's where the variable is defined.
So that might not work out fine. The only other option I can think of is to include a @categories variable in each method handling page routes.

ChigboIO commented on app/views/landing/_product.html.erb:
```diff
@@ -0,0 +1,9 @@
+	   <div class="card-desc center"><span><%= product[:text] %></span><br /><span class="price"><%= product[:price] %></span></div>
+	</a>
+</div>
+<div class= "col s12 m1 l0.1">&nbsp;</div>
```
Please, what does the class `10.1` do?

0duaht replied:
It gives a spacing of (0.1/12 \* width) of the div in which it is defined, while on a large screen.

ChigboIO commented on app/views/landing/index.html.erb:
```diff
@@ -0,0 +1,39 @@
+	<div class= "row container-trim-wide"><h5 class="cat-heading"><%= @categories[index].upcase %></h5></div>
+	<div class="container-large">
+		<div class= "row">
+			<%= render partial: "product", collection: products %>
```
You're putting every product in one row and separating them with spaces. This won't scale. Lol. There should be a better way man

0duaht replied:
I don't really get the "Lol" here. However, I'm not putting _every product_ in one row. I'm putting _each category_ in one row. All four products under that category are rendered in that row, while on a large screen.

ChigboIO commented:
Every file should end with new line.

Site is not responsive. Scatters on a mobile size.

I also feel like the products could have more spaces between them, and probably make some shadow on hover since we are using `materialize card`

Nice work in general man.

Commit:
Fix Rubocop/HoundCI violations
Why?
After pushing initially, issues were raised by HoundCI about following the Rubocop style guide specified.

This addresses the need by:
Fixing issues raised such as trailing white lines using the Sublime Linter package.

codesword commented on .project:
```diff
@@ -0,0 +1,12 @@
+	<natures>
+		<nature>org.radrails.rails.core.railsnature</nature>
+	</natures>
+</projectDescription>
```
Please add this file to .gitignore
